YouVsVirus/Assets/Scripts/LevelCollectMasks/hospital.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Components
+{
+    public class hospital : MonoBehaviour
+    {
+        // Start is called before the first frame update
+        void Start()
+        {
+            
+        }
+
+        // Update is called once per frame
+        void Update()
+        {
+            
+        }
+
+        void OnTriggerEnter2D (Collider2D other)
+        {
+            if (other.gameObject.tag == "Player") {
+                // The player is here. We check how many masks he has and
+                // give this option to the end level controller.
+                // The controller then decides to end the level or not.
+                Debug.Log ("Player entered hospital.");
+                // Get number of masks from player
+                int numberOfMasks = other.GetComponent<Player>().getNumMasks();
+                // Get end level controller and notify him about the number of masks
+                EndLevelControllerBase elc = LevelSettings.GetActiveEndLevelController();
+                elc.NotifyInt (numberOfMasks);
```

Comment:
Why have a variable elc here and not do it in one step?

Reply:
Because:
1. It does not affect performance (the compiler optimizes it out)
2. It is in my opinion more readable compared to LevelSettings.GetActiveEndLevelController().NotifyInt (numberOfMasks)